internal/crypto/libsodium/crypter.go

```diff
@@ -66,17 +78,58 @@ func (crypter *Crypter) setup() (err error) {
 	crypter.mutex.Lock()
 	defer crypter.mutex.Unlock()
 
-	if crypter.Key != "" {
-		return
+	keyString := crypter.KeyInline
+	if keyString == "" {
+		// read from file
+		var keyFileContents []byte
+		keyFileContents, err = ioutil.ReadFile(crypter.KeyPath)
+		if err != nil {
+			return
+		}
+
+		keyString = strings.TrimSpace(string(keyFileContents))
 	}
 
-	key, err := ioutil.ReadFile(crypter.KeyPath)
+	var rawKey []byte
+	if crypter.KeyTransform == "base64" {
```

Comment:
This definitely deserves decomposing to a separate file or at least function. Ideal way is to introduce a `LibsodiumKeyTransformer` interface, write different implementations (base64/hex/rpad-zero/no-transform) and separate constructor for this logic like `NewLibsodiumKeyTransformer(name string)`

Reply:
I moved the transform code into a separate file (`keytransform.go`). The transformations are now individual functions (`keyTransformBase64`,`keyTransformHex` and `keyTransformNone`).

Haven't implemented the interface approach yet. I presume if we would implement it like that, we'd change the
Crypter constructor from
```
CrypterFromKey(key string, keyTransform string)
```
to
```
CrypterFromKey(key string, keyTransform LibsodiumKeyTransformer)
```

This now also places the responsibility of creating a `LibsodiumKeyTransformer` instance to the caller. Creating a transform (i.e. using your suggested `NewLibsodiumKeyTransformer` function) may fail (by passing a unknown transformation name).
The caller (`ConfigureCrypter` in `internal/configure.go`) does not have error handling (i.e. cannot return an error), so you'd either have to
1) Panic()
2) Pass 'nil' as transform and check for nil in the crypter setup()
3) Pass a transform which always returns an error (which will return the error later during setup())

All of those options don't seem very clean. That's the reason why I haven't implemented the interface approach.